test/e2e/toolchaincluster_test.go

```diff
@@ -55,6 +61,26 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
 		util.ValidateKubeClient(t, targetClient, otherAwait.Namespace, &toolchainv1alpha1.ToolchainClusterList{})
 	})
 
+	t.Run("status is populated with info from kubeconfig", func(t *testing.T) {
+		if !tcStatusShouldBePopulated {
+			t.Skip("not applicable for this cluster type")
+			return
+		}
+		secret := corev1.Secret{}
+		require.NoError(t, await.Client.Get(context.TODO(), client.ObjectKey{Name: current.Spec.SecretRef.Name, Namespace: current.Namespace}, &secret))
+		apiConfig, err := clientcmd.Load(secret.Data["kubeconfig"])
+		require.NoError(t, err)
+
+		configContext, present := apiConfig.Contexts[apiConfig.CurrentContext]
+		require.True(t, present)
+
+		operatorNamespace := configContext.Namespace
+		apiEndpoint := apiConfig.Clusters[configContext.Cluster].Server
+
+		assert.Equal(t, operatorNamespace, current.Status.OperatorNamespace)
+		assert.Equal(t, apiEndpoint, current.Status.APIEndpoint)
+	})
```

Comment:
it would also maybe make sense to double check that it's the same as in the other fields:
```suggestion
		assert.Equal(t, current.Spec.APIEndpoint, current.Status.APIEndpoint)
		assert.Equal(t, current.Labels["namespace"], current.Status.OperatorNamespace)
	})
```

Reply:
I'm not sure this would bring much to the table since the spec field is going away and is not actually used for anything as soon as we start using kubeconfig for the client construction.